### PR TITLE
Refactor test helpers to use withr patterns

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

File: tests/testthat/test-cert.R [hunk 1]
```diff
@@ -1,149 +1,149 @@
...
+  addServer(
+    url = "https://localhost:4567/",
+    name = "cert_test_b",
+    quiet = TRUE
+  )
+
+  # add the certificate
+  addServerCertificate(
+    name = "cert_test_b",
+    certificate = "certs/sample.crt",
```
Comment (aronatkins) [inline tests/testthat/test-cert.R:35]: Another opportunity for `test_path`? (multiple examples)

**Review by aronatkins — APPROVED** [review]
There's a similar cleanup opportunity in `test-http.R`; the `datafile`, for example, is test-local (not shared state between the server and tests).

The `input` and `output` are different, unfortunately, and server-managed state. They can be cleaned up after each test, but also should be removed as the server shuts down.

**Comment by hadley** [comment]
I have my eye on `test-http.R` too, but I haven't quite figured out the right approach yet.

File: tests/testthat/test-cert.R (hunk 1 above)
Comment (hadley) [reply tests/testthat/test-cert.R:35]: Good point.

**Pull request merged**